cli/cmd/plugins/list/cmd.go

```diff
@@ -0,0 +1,359 @@
+	if err != nil {
+		return fmt.Errorf("getting output flag failed: %w", err)
+	}
+
+	pluginRegistryFlag, err := cmd.Flags().GetString(FlagRegistry)
+	if err != nil {
+		return fmt.Errorf("failed to get plugin registry flag: %w", err)
+	}
+
+	var pluginRegistries []string
+	if pluginRegistryFlag != "" {
+		regs := strings.Split(pluginRegistryFlag, ",")
+		pluginRegistries = append(pluginRegistries, regs...)
+	} else { //nolint:staticcheck // see TODOs below
+		// TODO: Load registries from config
+		// see https://github.com/open-component-model/ocm-project/issues/599
+
+		// TODO: Set default registry if no registry is provided
+		// see https://github.com/open-component-model/ocm-project/issues/598
+	}
+
+	// The config can contain several registries from which we want to list plugins
+	// e.g. registries:
+	//   - ghcr.io/open-component-model/plugin-registry:1.0.0
+	//   - example.com/ocm/plugin-registry:2.0.0
+	// Each registry (component) can contain several plugins (component references)
+	graph := dag.NewDirectedAcyclicGraph[string]()
+	var roots []string
+	for _, reg := range pluginRegistries {
+		logger.Debug("Getting plugin registry", "registry", reg)
+
+		ref, err := compref.Parse(reg)
+		if err != nil {
+			return fmt.Errorf("creating component reference for plugin registry %q failed: %w", reg, err)
+		}
+
+		repoProvider, err := ocm.NewComponentVersionRepositoryForComponentProvider(ctx, pluginManager.ComponentVersionRepositoryRegistry, credentialGraph, config, ref)
+		if err != nil {
+			return fmt.Errorf("could not initialize ocm repositoryProvider: %w", err)
+		}
+
+		repo, err := repoProvider.GetComponentVersionRepositoryForComponent(ctx, ref.Component, ref.Version)
+		if err != nil {
+			return fmt.Errorf("failed getting repository: %w", err)
+		}
+
+		var desc *descriptor.Descriptor
+
+		// Get latest version of plugin registry if no version is specified
+		if ref.Version == "" {
+			descs, err := ocm.GetComponentVersions(ctx, ocm.GetComponentVersionsOptions{
+				VersionOptions: ocm.VersionOptions{
+					LatestOnly: true,
+				},
+			}, ref.Component, ref.Version, repo)
+			if err != nil {
+				return fmt.Errorf("failed getting component versions for plugin registry: %w", err)
+			}
+
+			desc = descs[0]
+
+			if len(descs) == 0 {
+				return fmt.Errorf("no versions found for component %q in plugin registry", ref.Component)
+			}
+
+			// Add version to registry ref to be able to identify the source later
+			reg = fmt.Sprintf("%s:%s", reg, descs[0].Component.Version)
+		} else {
+			desc, err = repo.GetComponentVersion(ctx, ref.Component, ref.Version)
+			if err != nil {
+				return fmt.Errorf("failed getting component constructor for plugin registry: %w", err)
+			}
+		}
+
+		for _, r := range desc.Component.References {
+			logger.Debug("Adding plugin to graph", "plugin", r.Name, "version", r.Version, "registry", reg)
+
+			var info PluginInfo
+			for _, l := range r.Labels {
+				if l.Name == "ocm.software/pluginInfo" {
```

Comment:
Where is this value defined? If it's important/reusable, we should make a const out of it.

Reply:
It's fine if it's in this package if it's only used here.